rama-http/src/matcher/mod.rs

```diff
@@ -80,6 +84,8 @@ pub enum HttpMatcherKind<State, Body> {
     ///
     /// [`SocketAddr`]: std::net::SocketAddr
     Socket(SocketMatcher<State, Request<Body>>),
+    /// [`SubdomainTrieMatcher`], a matcher based on domain and subdomains using a trie structure.
+    SubdomainTrie(SubdomainTrieMatcher),
```

Comment:
Also add methods to `impl<State, Body> HttpMatcher<State, Body>` block: `any_subdomain`, `and_any_subdomain` and `or_any_subdomain`